components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/handler/impl/OAuth2AccessTokenHandler.java

```diff
@@ -104,52 +101,49 @@ protected AuthenticationResult doAuthenticate(MessageContext messageContext) {
                 contextParams[0] = contextParam;
                 requestDTO.setContext(contextParams);
 
-                OAuth2ClientApplicationDTO clientApplicationDTO = oAuth2TokenValidationService
-                        .findOAuthConsumerIfTokenIsValid
-                                (requestDTO);
-                OAuth2TokenValidationResponseDTO responseDTO = clientApplicationDTO.getAccessTokenValidationResponse();
+                OAuth2IntrospectionResponseDTO oAuth2IntrospectionResponseDTO =
+                        oAuth2TokenValidationService.buildIntrospectionResponse(requestDTO);
 
-                if (!responseDTO.isValid()) {
+                if (!oAuth2IntrospectionResponseDTO.isActive()) {
                     return authenticationResult;
                 }
 
-                if (!isTokenBindingValid(messageContext, responseDTO.getTokenBinding(),
-                        clientApplicationDTO.getConsumerKey(), accessToken)) {
+                TokenBinding tokenBinding = new TokenBinding(oAuth2IntrospectionResponseDTO.getBindingType(),
+                        oAuth2IntrospectionResponseDTO.getBindingReference());
+                if (!isTokenBindingValid(messageContext, tokenBinding,
+                        oAuth2IntrospectionResponseDTO.getClientId(), accessToken)) {
                     return authenticationResult;
                 }
 
                 authenticationResult.setAuthenticationStatus(AuthenticationStatus.SUCCESS);
 
-                if (StringUtils.isNotEmpty(responseDTO.getAuthorizedUser())) {
-                    User user = new User();
-                    String tenantAwareUsername =
-                            MultitenantUtils.getTenantAwareUsername(responseDTO.getAuthorizedUser());
-                    user.setUserName(UserCoreUtil.removeDomainFromName(tenantAwareUsername));
-                    user.setUserStoreDomain(UserCoreUtil.extractDomainFromName(tenantAwareUsername));
-                    user.setTenantDomain(MultitenantUtils.getTenantDomain(responseDTO.getAuthorizedUser()));
-                    authenticationContext.setUser(user);
+                User authorizedUser = oAuth2IntrospectionResponseDTO.getAuthorizedUser();
```

Comment:
Can we check the behaviour for client credentials grant?

I mean if an application gets a token using cc and attempts to authenticate the authenticated user here will be the app developer. But just wondering that is the correct expectation?